boomer/src/main.rs

```diff
@@ -43,14 +43,21 @@ struct CliArgs {
     verbose: u8,
 }
 
-fn find_connector(device: &Device, connector_name: Option<&str>) -> Option<Rc<Connector>> {
+fn find_connectors(device: &Device, connector_name: Option<&str>) -> Vec<Rc<Connector>> {
```

Comment:
I don't think we need to allocate a Vec here to store the result, we only care about a single one anyway.

I guess we could return a `Result<Rc<Connector>>` and fail if we have no connector, or more than one.